apps/eaas-api/src/app/controllers/auth.controller.ts

```diff
@@ -1,150 +1,87 @@
 import { Request, Response } from "express";
 import { authService, tokenService, userService } from "../services";
 import { envConfig } from "../../../src/config/env.config";
-import { handleApiError } from "../utils/errors";
+
+import { emailService } from "../services/email.service";
+import { EaasUser, RequestWithUser } from "src/types/EaasUser";
 
 const verifyToken = async (req: Request, res: Response) => {
-  try {
-    const { token } = req.body;
-    await tokenService.verifyToken(token, envConfig.jwt.secret);
-    res.status(200).json();
-  } catch (error) {
-    console.error(error);
-    res.status(400).json({ error: "this invitation is invalid" });
-  }
+  const { token } = req.body;
+  await tokenService.verifyToken(token, envConfig.jwt.secret);
+  res.status(200).json();
 };
 
 const register = async (req: Request, res: Response) => {
-  try {
-    const { user, inviteToken } = req.body;
-    const { name, password, phone } = user;
-    const { user: dbUser } = await userService.createUserWithInvite({
-      name,
-      password,
-      phone,
-      inviteToken,
-    });
+  const { user, inviteToken } = req.body;
+  const { password } = user;
+  const { user: dbUser } = await userService.createUserWithInvite({
+    password,
+    inviteToken,
+  });
 
-    if (!dbUser || !dbUser.id) {
-      res.status(400).json({ error: "could not create user" });
-    } else {
-      const tokens = await tokenService.generateAuthTokens(dbUser.id);
-      res.status(201).json({
-        user: dbUser,
-        tokens,
-      });
-    }
-  } catch (error) {
-    handleApiError(res, error);
+  if (!dbUser || !dbUser.id) {
+    res.status(400).json({ error: "could not create user" });
```

Comment:
What would be the case(s) in which this fails?

Reply:
If the `createUserWithInvite` fails, it will throw an error, so don't think it would hit this conditional.